ProcessLib/SmallDeformation/SmallDeformationFEM.h

```diff
@@ -0,0 +1,335 @@
+        {
+            _ip_data.emplace_back(*_process_data._material);
+            auto& ip_data = _ip_data[ip];
+            ip_data._detJ = shape_matrices[ip].detJ;
+            ip_data._b_matrices.resize(
+                KelvinVectorDimensions<DisplacementDim>::value,
+                ShapeFunction::NPOINTS * DisplacementDim);
+            LinearBMatrix::computeBMatrix<
+                DisplacementDim, ShapeFunction::NPOINTS,
+                typename ShapeMatricesType::GlobalDimNodalMatrixType,
+                BMatrixType>(shape_matrices[ip].dNdx, ip_data._b_matrices);
+
+            ip_data._sigma.resize(
+                KelvinVectorDimensions<DisplacementDim>::value);
+            ip_data._sigma_prev.resize(
+                KelvinVectorDimensions<DisplacementDim>::value);
+            ip_data._eps.resize(KelvinVectorDimensions<DisplacementDim>::value);
+            ip_data._eps_prev.resize(
+                KelvinVectorDimensions<DisplacementDim>::value);
+            ip_data._C.resize(KelvinVectorDimensions<DisplacementDim>::value,
+                              KelvinVectorDimensions<DisplacementDim>::value);
+
+            _secondary_data.N[ip] = shape_matrices[ip].N;
+        }
+    }
+
+    void assembleConcrete(
+        double const t, std::vector<double> const& local_x,
+        NumLib::LocalToGlobalIndexMap::RowColumnIndices const& indices,
+        GlobalMatrix& /*M*/, GlobalMatrix& /*K*/, GlobalVector& b) override
+    {
+        _localRhs.setZero();
+
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+
+        NodalDisplacementVectorType local_displacement(ShapeFunction::NPOINTS *
+                                                       DisplacementDim);
+
+        SpatialPosition x_position;
+        x_position.setElementID(_element.getID());
+
+        for (unsigned ip = 0; ip < n_integration_points; ip++)
+        {
+            x_position.setIntegrationPoint(ip);
+            auto const& wp = _integration_method.getWeightedPoint(ip);
+            auto const& detJ = _ip_data[ip]._detJ;
+
+            auto const& B = _ip_data[ip]._b_matrices;
+            auto const& eps_prev = _ip_data[ip]._eps_prev;
+            auto const& sigma_prev = _ip_data[ip]._sigma_prev;
+
+            auto& eps = _ip_data[ip]._eps;
+            auto& sigma = _ip_data[ip]._sigma;
+            auto& C = _ip_data[ip]._C;
+            auto& material_state_variables =
+                *_ip_data[ip]._material_state_variables;
+
+            eps.noalias() =
+                B *
+                Eigen::Map<typename BMatricesType::NodalForceVectorType const>(
+                    local_x.data(), ShapeFunction::NPOINTS * DisplacementDim);
+
+            _ip_data[ip]._solid_material.computeConstitutiveRelation(
+                t, x_position, _process_data.dt, eps_prev, eps, sigma_prev,
+                sigma, C, material_state_variables);
+
+            _localRhs.noalias() -=
```

Comment:
@nagelt What I don't understand is inconsistency of signs in residuals and Jacobian. if the residual is given as `r = external force - inner force`, Jacobian should be give as `J=dr/du=- d(inner force)/dt` with a negative sign. 

but if `_localRhs` is actually used as `-r`, the expression here is correct. I thought the assembly should return `r` and not `-r`.